clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3080,12 +3083,33 @@ private int processKafkaDataMessage(
         // update checksum for this PUT message if needed.
         partitionConsumptionState.maybeUpdateExpectedChecksum(keyBytes, put);
 
-        // Do transorfmation recompute key, value and partition
+        // Check if put.getSchemaId is positive, if not default to 1
+        int putSchemaId = put.getSchemaId() > 0 ? put.getSchemaId() : 1;
+
+        // Do transformation recompute key, value and partition
         if (recordTransformer != null) {
+          ByteBuffer valueBytes = put.getPutValue();
+          Schema valueSchema = schemaRepository.getValueSchema(storeName, putSchemaId).getSchema();
+
+          // Decompress/assemble record
+          Object assembledObject = chunkAssembler.bufferAndAssembleRecord(
+              consumerRecord.getTopicPartition(),
+              putSchemaId,
+              keyBytes,
+              valueBytes,
+              consumerRecord.getOffset(),
+              Lazy.of(() -> new AvroGenericDeserializer<>(valueSchema, valueSchema)),
+              putSchemaId,
+              compressorFactory.getCompressor(compressionStrategy, kafkaVersionTopic));
+
+          // Current record is a chunk. We only write to the storage engine for fully assembled records
+          if (assembledObject == null) {
+            break;
```

Comment:
I don't think is the right thing to do here, I think return; would be good.

Reply:
What value should I return here, since the return type for this function is an int?

Reply:
I think it'd be fine to return zero since the return value is what was committed to storage (we didn't write anything to storage at this stage).